euphonic/structure_factor.py

```diff
@@ -106,6 +108,41 @@ def __setattr__(self, name, value):
                 ureg(getattr(self, name)).to(value)
         super(StructureFactor, self).__setattr__(name, value)
 
+    def calculate_1d_average(self,
```

Comment:
How would users use this? Again, would they have to call this for different `StructureFactor` objects with sampling at different |Q| to build their own 2D |Q|,E spectrum?

Reply:
It's already called by `euphonic.powder.sample_sphere_structure_factor`, and could be used by other workflow-specific sampling functions we develop.

As a fundamental building-block for anyone who wants to play around with obtaining 1D spectra from different q-sampling schemes, it makes sense to provide public access on the StructureFactor object. It is really _the_ transformation from StructureFactor to Spectrum1D, just as `calculate_sqw_map` is the transformation from StructureFactor to Spectrum2D. Perhaps they should have more similar names.

Reply:
`calculate_sw_spectrum` would probably be a similar name, but without the formatting of S(ω) it looks pretty strange...

Reply:
I guess it could be something like `calculate_energy_spectrum` but I don't really find that less confusing!

Reply:
True, I'd be happy to leave this for now, we can always deprecate and rename in the future, maybe it'll become clearer what it should be called as we actually start to use it and add more powdery things

Reply:
I suppose it could be given a "private" name if we don't want to commit this name to the public API yet?

Reply:
It's probably ok to leave it for now, if we don't make it public now we might never get round to it!